lib/OpenQA/Parser/Format/IPA.pm

```diff
@@ -37,14 +38,35 @@ sub parse {
     foreach my $res (@{$decoded_json->{tests}}) {
         my $result = {};
         my $t_name = $res->{name};
-        $t_name =~ s/\[\w+:\/\/(\d+\.){3}\d+//;
-        $t_name =~ s/\]$//;
+
+        if ($t_name =~ /^(?<path>[\w\/]+\/)?(?<file>\w+)\.py::(?<method>\w+)\[\w+:\/\/(\d+\.){3}\d+(-(?<param>.+))?\]$/)
+        {
+            $t_name = '';
+            $t_name .= $+{path} if ($+{path});
+            $t_name .= $+{file};
+            $t_name .= '_' . $+{method} if ($+{file} ne $+{method});
+            if ($+{param}) {
+                my $param = $+{param};
+                $param =~ s/\.service$//;
+                $t_name .= '_' . $param;
+            }
+        }
+
+        # If a tests was triggered twice, we need to unique the name
+        if (exists($unique_names{$t_name})) {
+            $t_name .= '_' . sprintf("%02d", ++$unique_names{$t_name});
```

Comment:
I'd move the '_' into the `sprintf`-string.

Reply:
thx, done